auth.go

```diff
@@ -50,5 +50,13 @@ func authorized(conn tls.ConnectionState) bool {
 		}
 	}
 
+	for _, clientSAN := range cert.DNSNames {
```

Comment:
How else you wanna do it tho? I'm kinda assuming the list of SANs in the cert won't be huge, so looping over them should be fine.